src/Features/LensEffects.cpp

```diff
@@ -0,0 +1,1110 @@
+		ImGui::SliderFloat("Max Expansion", &mainSettings.HL_ExpMaxSize, 0.2f, 1.0f);
+		ImGui::SliderFloat("Rotation Speed", &mainSettings.HL_RotationSpeed, 0.0f, 1.0f);
+		ImGui::SliderFloat("Rake Increment (degrees)", &mainSettings.HL_LineVolume, 1.0f, 45.0f, "%.0f");
+		ImGui::SliderFloat("Length", &mainSettings.HL_LineLength, 0.05f, 0.8f);
+		ImGui::SliderFloat("Width", &mainSettings.HL_LineWidth, 0.03f, 0.5f);
+		ImGui::SliderFloat("Taper", &mainSettings.HL_LineTaper, 0.0f, 0.15f);
+		ImGui::SliderFloat("Chromatic Shift", &mainSettings.HL_ColorShift, 0.0f, 1.0f);
+		ImGui::Spacing();
+		ImGui::Spacing();
+
+		ImGui::Text("Color ");
+		ImGui::SameLine();
+
+		float4& Color = settings.coldsettings.HL_Color;
+		if (ImGui::ColorButton("Pick", ImVec4(Color.x, Color.y, Color.z, Color.w)))
+			ImGui::OpenPopup("ColorPopup_");
+
+		if (ImGui::BeginPopup("ColorPopup_", ImGuiWindowFlags_NoMove)) {
+			ImGui::SameLine(ImGui::GetContentRegionAvail().x - ImGui::GetFontSize() + ImGui::GetStyle().ItemInnerSpacing.x * 2);
+			if (ImGui::SmallButton("X"))
+				ImGui::CloseCurrentPopup();
+			ImGui::ColorPicker4("RGBA", &Color.x);
+			ImGui::EndPopup();
+		}
+		ImGui::TreePop();
+	}
+	ImGui::Spacing();
+	ImGui::Spacing();
+	ImGui::Spacing();
+
+	//// Lens Ice ////
+	ImGui::SeparatorText("Lens Ice");
+
+	if (upscalingActive)
+		ImGui::BeginDisabled();
+
+	ImGui::Spacing();
+	ImGui::SliderFloat("Ice Intensity", &mainSettings.LI_Intensity, 0.01f, 2.0f);
+	ImGui::SliderFloat("Fade Duration", &mainSettings.LI_FadeDuration, 1.0f, 30.0f, "%.0f");
+	ImGui::SliderFloat("Fade In Delay", &coldSettings.LI_FadeIn, 0.3f, 0.95f);
+	ImGui::SliderFloat("Fade Out Delay", &coldSettings.LI_FadeOut, 0.0f, 0.7f);
+	ImGui::Spacing();
+	ImGui::Spacing();
+	ImGui::Text("Color ");
+	ImGui::SameLine();
+
+	float4& Color = settings.coldsettings.LI_Color;
+	if (ImGui::ColorButton("Pick ##LI", ImVec4(Color.x, Color.y, Color.z, Color.w)))
+		ImGui::OpenPopup("ColorPopup_ ##LI");
+
+	if (ImGui::BeginPopup("ColorPopup_ ##LI", ImGuiWindowFlags_NoMove)) {
+		ImGui::SameLine(ImGui::GetContentRegionAvail().x - ImGui::GetFontSize() + ImGui::GetStyle().ItemInnerSpacing.x * 2);
+		if (ImGui::SmallButton("X"))
+			ImGui::CloseCurrentPopup();
+		ImGui::ColorPicker4("RGBA ##LI", &Color.x);
+		ImGui::EndPopup();
+	}
+	ImGui::Spacing();
+	ImGui::Spacing();
+	ImGui::Spacing();
+
+	if (upscalingActive)
+		ImGui::EndDisabled();
+
+	//// Chromatic Aberration ////
+	ImGui::SeparatorText("Chromatic Aberration");
+
+	ImGui::Spacing();
+	ImGui::SliderFloat("CA Intensity", &mainSettings.CA_Intensity, 0.0f, 3.0f);
+	ImGui::SliderFloat("CA Motion Threshold", &mainSettings.CA_Threshold, 0.001f, 0.1f);
+	ImGui::SliderFloat("CA Max Offset", &mainSettings.CA_MaxOffset, 0.001f, 0.015f);
+	ImGui::Checkbox("Use Old School CA", &coldSettings.useOldSchoolCA);
+	ImGui::Spacing();
+	ImGui::Spacing();
+	ImGui::Spacing();
+
+	//ImGui::SeparatorText("Lens Rain Effect");
```

Comment:
Don't commit dead code. It adds a maintenance burden and doesn't guarantee to work even when uncommented particularly as the feature changes.